src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/X509Certificate2Extensions.cs

```diff
@@ -23,5 +27,20 @@ public static bool SupportsUse(this X509Certificate2 certificate, X509KeyUsageFl
 
             return certificate.Extensions.OfType<X509KeyUsageExtension>().FirstOrDefault()?.KeyUsages.HasFlag(use) ?? true;
         }
+
+        /// <summary>
+        /// Tests whether a certificate supports server authentication.
+        /// </summary>
+        /// <param name="certificate">Certificate to test.</param>
+        /// <returns>True if the certificate supports server authentication; false otherwise.</returns>
+        public static bool SupportsServerAuthentication(this X509Certificate2 certificate)
+        {
+            var oidValues =
+                from extension in certificate.Extensions.OfType<X509EnhancedKeyUsageExtension>()
+                from oid in extension.EnhancedKeyUsages.OfType<Oid>()
+                select oid.Value;
+
+            return oidValues.Any(oid => oid.Equals(ServerAuthenticationOid, StringComparison.Ordinal));
```

Comment:
Any possibility of a **null** value causing `oid.Equals()` to throw an `ArgumentNullException`?

Reply:
Probably not, but as it's _possible_ to construct an Oid with `null` value I'll add a check to guard against that.

Reply:
Need only switch around the test, using static `Equals()`:

```suggestion
            return oidValues.Any(oid => string.Equals(oid, ServerAuthenticationOid, StringComparison.Ordinal));
```